src/ui/simulator/toolbox/components/notebook/notebook.cpp

```diff
@@ -1023,6 +1023,18 @@ Notebook::Page* Notebook::find(const wxString& name, bool warn)
     return nullptr;
 }
 
+void Notebook::hide_page(const wxString& name)
+{
+    Page* page = find(name);
+    page->visible(false);
+}
+
+void Notebook::show_page(const wxString& name)
+{
+    Page* page = find(name);
+    page->visible(true);
+}
+
```

Comment:
Suggestion : use a slightly different interface
```cpp
void Notebook::setVisibility(const wxString& name, bool visible)
{
    Page* page = find(name);
    page->visible(visible);
}
```
Also, handle the case where `find` does not find, e.g by calling `return;` with an error in the logs.

Reply:
```cpp
Notebook::Page* Notebook::find(const wxString& name, bool warn)
{
    foreach (auto* page, pPages)
    {
        if (page->name() == name)
            return page;
    }

#ifndef NDEBUG
    if (warn)
    {
        String sname;
        wxStringToString(name, sname);
        logs.warning() << "notebook: impossible to find '" << sname << "'";
    }
#endif
    return nullptr;
}
```

In debug mode, a warning is issued if find does not find, so it may not be necessary to display an error in the logs.